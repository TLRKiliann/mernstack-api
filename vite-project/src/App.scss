#root {
  width: 100%;
  height: auto;
  background: #424242;
}
