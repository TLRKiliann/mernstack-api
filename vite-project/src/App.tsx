import React from 'react'
import { Routes, Route } from 'react-router-dom'
import PrivateRoute from './PrivateRoute'
import NavBar from './components/NavBar'
import Login from './pages/Login'
import Subscribe from './pages/Subscribe'
import Home from './pages/Home'
import Profile from './pages/Profile'
import Online from './pages/Online'
import ChatComputer from './components/ChatComputer'
import ComputerRoom from './components/ComputerRoom'
import PrivateMessage from './components/privatechat/PrivateMessage'
import PageNotFound from './pages/PageNotFound'
import './App.scss'

const App: React.FC = () => {
  return(
    <div>
      <NavBar />
      <Routes>
        <Route exact path='/' element={<Home />} />
        <Route exact path='/subscribe' element={<Subscribe />} />
        <Route exact path='/login' element={<Login />} />
        <Route path='/profile' element={
          <PrivateRoute>
            <Profile />
          </PrivateRoute>
        } />
        <Route path='/online' element={
          <PrivateRoute>
            <Online />
          </PrivateRoute>
        } />
        <Route path='/chatcomputer/:id' element={
          <PrivateRoute>
            <ChatComputer />
          </PrivateRoute>
        } />
        <Route path='/computerroom/:link' element={
          <PrivateRoute>
            <ComputerRoom />
          </PrivateRoute>
        } />
        <Route path='/computerroom/privatechat/:retrieveRoom' element={
          <PrivateRoute>
            <PrivateMessage />
          </PrivateRoute>
        } />
        <Route exact path='*' element={<PageNotFound />} />
      </Routes>
    </div>
  )
}

export default App