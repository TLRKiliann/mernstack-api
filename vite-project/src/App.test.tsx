//import React from "react";
import {beforeEach, describe, expect, test, it, vi} from 'vitest';
import {fireEvent, render, screen, waitFor} from '@testing-library/react';
//import "@testing-library/jest-dom/extend-expect"
//import "@testing-library/jest-dom";
import { act } from 'react-dom/test-utils';
import {create} from 'react-test-renderer';
import App from "./App.tsx";

test('MatchSnapShot test App 1', () => {
	const tree = create(<App />)
	expect(tree.toJSON()).toMatchSnapshot();
})