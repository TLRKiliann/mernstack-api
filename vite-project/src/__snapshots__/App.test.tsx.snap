// Vitest Snapshot v1

exports[`MatchSnapShot test App 1 1`] = `
<div
  className="App"
>
  <h1>
    Notes MERN-stack API
  </h1>
  <div
    style={
      {
        "border": "1px solid orange",
      }
    }
  >
    <div>
      Hello First ! 
      Jess
    </div>
    <div>
      1 - Hello Last ! 
      Fruital
      <div
        style={
          {
            "color": "white",
          }
        }
      >
        2 - SubComponent subLast: 
        Fruital
      </div>
    </div>
    <div>
      Hello Age ! 
      42
    </div>
    <div>
      <label>
        1 - Hello LocalNote ! 
        Highland
      </label>
      <button
        onClick={[Function]}
        style={
          {
            "marginLeft": "10px",
          }
        }
      >
        Abbr location
      </button>
      <div
        style={
          {
            "border": "1px solid lightgreen",
            "color": "lightgreen",
          }
        }
      >
        2 - SubComponent subLocation: 
        Highland
      </div>
    </div>
  </div>
  <div
    style={
      {
        "border": "1px solid orange",
      }
    }
  >
    <div>
      Hello First ! 
      Celestine
    </div>
    <div>
      1 - Hello Last ! 
      NeverMind
      <div
        style={
          {
            "color": "white",
          }
        }
      >
        2 - SubComponent subLast: 
        NeverMind
      </div>
    </div>
    <div>
      Hello Age ! 
      33
    </div>
    <div>
      <label>
        1 - Hello LocalNote ! 
        Kentuky
      </label>
      <button
        onClick={[Function]}
        style={
          {
            "marginLeft": "10px",
          }
        }
      >
        Abbr location
      </button>
      <div
        style={
          {
            "border": "1px solid lightgreen",
            "color": "lightgreen",
          }
        }
      >
        2 - SubComponent subLocation: 
        Kentuky
      </div>
    </div>
  </div>
  <div
    style={
      {
        "border": "1px solid orange",
      }
    }
  >
    <div>
      Hello First ! 
      Ness
    </div>
    <div>
      1 - Hello Last ! 
      Namibi
      <div
        style={
          {
            "color": "white",
          }
        }
      >
        2 - SubComponent subLast: 
        Namibi
      </div>
    </div>
    <div>
      Hello Age ! 
      22
    </div>
    <div>
      <label>
        1 - Hello LocalNote ! 
        Zeland
      </label>
      <button
        onClick={[Function]}
        style={
          {
            "marginLeft": "10px",
          }
        }
      >
        Abbr location
      </button>
      <div
        style={
          {
            "border": "1px solid lightgreen",
            "color": "lightgreen",
          }
        }
      >
        2 - SubComponent subLocation: 
        Zeland
      </div>
    </div>
  </div>
</div>
`;
